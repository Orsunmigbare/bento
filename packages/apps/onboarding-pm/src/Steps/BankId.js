import React from "react"
import { Wrapper, Heading, Button } from "@staccx/base"

const BankId = ({ steps }) => (
  <Wrapper size="small">
    <Heading level="2">BankID-loggin</Heading>
    <Button variant="primary" onClick={() => steps.consent()}>
      Neste steg
    </Button>
  </Wrapper>
)

export default BankId
