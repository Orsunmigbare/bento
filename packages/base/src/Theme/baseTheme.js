import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
