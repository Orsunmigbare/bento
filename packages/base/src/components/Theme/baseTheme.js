export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
