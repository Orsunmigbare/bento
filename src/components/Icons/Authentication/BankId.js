import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

const BankId = ({ className }) => (
  <SVG
    width="16"
    height="16"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect y="3" width="5" height="2" />
    <rect x="11" y="3" width="5" height="2" />
    <rect x="11" y="6" width="5" height="2" />
    <rect x="6" y="6" width="4" height="2" />
    <rect y="9" width="5" height="2" />
    <rect x="6" y="9" width="4" height="2" />
    <rect y="12" width="5" height="2" />
    <rect x="11" y="12" width="5" height="2" />
  </SVG>
)

const SVG = styled.svg`
  fill: currentColor;
`

BankId.propTypes = {
  className: PropTypes.string
}

export default BankId
