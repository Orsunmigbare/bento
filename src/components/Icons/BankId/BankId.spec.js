import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import BankId from "./BankId"

describe("BankId", () => {
  describe("Snapshots", () => {
    it("BankId", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <BankId />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
