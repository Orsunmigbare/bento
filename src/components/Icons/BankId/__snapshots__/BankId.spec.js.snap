// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankId Snapshots BankId 1`] = `
<svg
  height="16"
  viewBox="0 0 16 16"
  width="16"
  xmlns="http://www.w3.org/2000/svg"
>
  <rect
    height="2"
    width="5"
    y="3"
  />
  <rect
    height="2"
    width="5"
    x="11"
    y="3"
  />
  <rect
    height="2"
    width="5"
    x="11"
    y="6"
  />
  <rect
    height="2"
    width="4"
    x="6"
    y="6"
  />
  <rect
    height="2"
    width="5"
    y="9"
  />
  <rect
    height="2"
    width="4"
    x="6"
    y="9"
  />
  <rect
    height="2"
    width="5"
    y="12"
  />
  <rect
    height="2"
    width="5"
    x="11"
    y="12"
  />
</svg>
`;
